tools/spatialDraw/index.js

```diff
@@ -0,0 +1,324 @@
+        appActive = true;
+        scene.visible = true;
+    }
+});
+envelope.onClose(() => {
+    drawingManager.disableInteractions();
+    appActive = false;
+    scene.visible = false;
+    location.reload();
+});
+
+function resizeText() {
+    text.innerText = text.innerText.toUpperCase();
+    const fontSize = Math.min(70, (((text.innerText.length * 7) + 500) / (text.innerText.length))); // font size increases up to 45pt
+    text.style.fontSize = fontSize + 'pt';
+}
+resizeText();
+
+function resetScroll() {
+    if (window.scrollX !== 0 || window.scrollY !== 0) {
+        window.scrollTo(0, 0); // don't let keyboard events scroll the window
+    }
+    parent.postMessage(JSON.stringify({resetScroll: true}), '*');
+}
+resetScroll();
+
+function initDrawingApp() {
+    drawingManager = new DrawingManager(mainContainerObj, camera);
+    drawingManager.addCallback('update', drawingData => {
+        drawingData.time = Date.now();
+        spatialInterface.writePublicData('storage', 'drawing', drawingData);
+    });
+
+    document.addEventListener('pointerdown', e => {
+        if (e.button === 0) {
+            drawingManager.onPointerDown(e);
+        }
+    });
+    document.addEventListener('pointermove', e => {
+        drawingManager.onPointerMove(e);
+    });
+    document.addEventListener('pointerup', e => {
+        if (e.button === 0) {
+            drawingManager.onPointerUp(e);
+        }
+    });
+    if (loadedDrawing) {
+        drawingManager.deserializeDrawing(loadedDrawing);
+        loadedDrawing = null;
+    }
+    drawingManager.enableInteractions();
+
+    drawingManager.addCallback('size', (size) => {
+        sizeCircles.forEach(sizeCircle => sizeCircle.classList.remove('active'));
+        const nextCircle = sizeCircles.find(circle => circle.dataset.size === size);
+        nextCircle.classList.add('active'); // Activates
+    });
+    drawingManager.addCallback('color', (color) => {
+        sizeCircles.forEach(sizeCircle => sizeCircle.setColor(color));
+        colorCircles.forEach(colorCircle => colorCircle.classList.remove('active'));
+        colorCircles.find(circle => circle.dataset.color === color).classList.add('active');
+    });
+    drawingManager.addCallback('eraseMode', (eraseMode) => {
+        if (eraseMode) {
+            eraseCircle.classList.add('active');
+        } else {
+            eraseCircle.classList.remove('active');
+        }
+    });
+    drawingManager.addCallback('cursor', cursor => {
+        cursorMenuOptions.forEach(option => option.classList.remove('active'));
+        cursorMenuOptions.find(option => cursor === drawingManager.cursorMap[option.dataset.cursor]).classList.add('active');
+    });
+
+    initializedApp = true;
+}
+
+let mainData = {
+    width: 0,
+    height: 0
+};
+
+// eslint-disable-next-line no-undef
+main = ({width, height}) => {
+    mainData.width = width;
+    mainData.height = height;
+};
+
+let rendererStarted = false;
+
+function initRenderer() {
```

Comment:
Not sure if it's worth integrating yet but a potential simplification of the boilerplate Three.js scene instantiation and rendering with the gl-proxy is to use the ThreejsInterface that @hobinjk-ptc wrote (example: https://github.com/ptcrealitylab/vuforia-spatial-analytics-addon/blob/map-tool/tools/map/index.js)